src/c_datove_typy.md

```diff
@@ -1,6 +1,127 @@
+jednoduššeji přemýšlelo. Pokud programujeme textový editor, chceme se bavit o znacích, odstavcích,
+fontech či barvách, pokud programujeme počítačovou hru, chceme se bavit o zbraních, brnění, kouzlech
+či pixelech.
+
+Přesně k tomu slouží **datové typy**, které popisují, jak budeme interpretovat konkrétní hodnoty
+daného typu v paměti, kolik bytů budou zabírat a jaké operace nad nimi budeme moct provádět. Jazyk C
+obsahuje několik vestavěných datových typů, [později](c_struktury.md) si ukážeme, jak vytvořit
+své vlastní.
+
+### Celočíselné datové typy
+Asi nejpřirozenějším a nejpoužívanějším datovým typem ve většině programovacích jazyků jsou (celá)
+čísla. Tyto číselné datové typy nám umožňují pracovat s celými čísly, které mají typicky jednotky
+(1 - 8) bytů[^1]. Počet bytů udává, jak velký rozsah můžou hodnoty daného typu obsahovat. Například
+číslo s 2 byty (16 bity) bez znaménka může obsahovat hodnoty 0 až 2<sup>16</sup>-1. Čím více bytů,
+tím více zabere hodnota daného typu místa v paměti.
+
+[^1]: I když 8 bytů (64 bitů) může znít jako málo, tak pomocí takového čísla můžeme vyjádřit 2<sup>64</sup>
+(neboli `18 446 744 073 709 551 616`) různých hodnot, což pro naprostou většinu běžného použití čísel
+bohatě stačí.
+
+U celých číselných typů se rozlišuje, zda jsou **signed** (se znaménkem) nebo **unsigned** (bez
+znaménka, nezáporné). Tato vlastnost udává, jaký rozsah hodnot můžou hodnoty daného typu nabývat
+(tj. jestli můžou být i záporné nebo ne). Například číslem o velikosti jednoho bytu můžeme
+reprezentovat 256 různých hodnot. Pokud budeme interpretovat toto číslo se znaménkem, tak může uchovávat
+hodnoty -128 až 127. Pokud ho budeme interpretovat bez znaménka, tak může uchovávat hodnoty 0 až 255.
+
+C obsahuje několik základních typů celočíselných proměnných, které se liší v tom, kolik mají bytů a
+jestli jsou znaménkové nebo ne. Pokud před název typu napíšeme `signed`, bude se jednat o znaménkový
+typ, pokud použijeme `unsigned`, tak použijeme typ bez znaménka. Většina typů je implicitně se
+znaménkem, tj. `int` je to samé jako `signed int`. V následující tabulce je seznam nejčastějších
+celočíselných typů[^2]:
+
+| Název | Počet bytů | Rozsah hodnot | Se znaménkem |
+|---|:---:|:---:|:---:|
+| `char` nebo<br />`signed char` | 1 | [-128; 127] | <i class="fa fa-check"></i> |
+| `unsigned char` | 1 | [0; 255] | <i class="fa fa-times"></i> |
+| `short` nebo<br />`signed short` | 2 | [-32 768; 32 767] | <i class="fa fa-check"></i> |
+| `unsigned short` | 2 | [0; 65 535] | <i class="fa fa-times"></i> |
+| **`int`** nebo<br />`signed int` | 4 | [-2 147 483 648; 2 147 483 647] | <i class="fa fa-check"></i> |
+| `unsigned int` | 4 | [0; 4 294 967 295] | <i class="fa fa-times"></i> |
+| `long` nebo<br />`signed long` | 8 | [-9 223 372 036 854 775 808;<br />9 223 372 036 854 775 807] | <i class="fa fa-check"></i> |
+| `unsigned long` | 8 | [0; 18 446 744 073 709 551 615] | <i class="fa fa-times"></i> |
```

Comment:
mozna bych nejak zminil typy `uint32_t`, `uint64_t` atd? lepsi nez si hrat s long long atd...

Reply:
Hmm, zase to nechci mít tak super detailní. Podobné věci by to asi chtělo dát do nějaké "advanced" sekce. A 8bytové inty snad nebudou během semestru potřeba.